pkg/asset/installconfig/aws/validation.go

```diff
@@ -351,6 +380,40 @@ func validateRegion(region string) error {
 	return validateEndpointAccessibility(ec2Session.Endpoint)
 }
 
+func validateZoneLocal(ctx context.Context, meta *Metadata, fldPath *field.Path, zoneName string) *field.Error {
+	sess, err := meta.Session(ctx)
+	if err != nil {
+		return field.Invalid(fldPath, zoneName, fmt.Sprintf("unable to start a session: %s", err.Error()))
+	}
+	zones, err := describeFilteredZones(ctx, sess, meta.Region, []string{zoneName})
+	if err != nil {
+		return field.Invalid(fldPath, zoneName, fmt.Sprintf("unable to get describe zone: %s", err.Error()))
+	}
+	validType := false
+	groupOptedIn := false
+	groupName := ""
+	for _, zone := range zones {
+		if aws.StringValue(zone.ZoneName) == zoneName {
+			if aws.StringValue(zone.ZoneType) == awstypes.LocalZoneType {
+				validType = true
+			}
+			groupName = aws.StringValue(zone.GroupName)
+			if aws.StringValue(zone.OptInStatus) == "opted-in" {
+				groupOptedIn = true
+			}
+		}
+	}
+	// validate if zone group has been opted-in
+	if !groupOptedIn && len(zones) > 0 {
+		return field.Invalid(fldPath, zoneName, fmt.Sprintf("zone group is not opted-in: %s", groupName))
+	}
+	// validate the name
+	if !validType {
+		return field.Invalid(fldPath, zoneName, fmt.Sprintf("invalid local zone name: %s", zoneName))
+	}
+	return nil
```

Comment:
This part must be improved to cover when non-local-zones has been added. Maybe returning directly inside the loop would be a better approach to get this example:
~~~
compute:
- name: worker
  platform:
    aws:
      zones:
      - us-east-1a
- name: edge
  platform:
    aws:
      zones:
      - 'us-east-1a'
~~~

Reply:
[resolved](https://github.com/openshift/installer/pull/7137#issuecomment-1624340777)